sdks/java/io/hdfs/pom.xml

```diff
@@ -58,12 +58,41 @@
     </dependency>
 
     <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-annotations</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+    </dependency>
+
+    <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-client</artifactId>
       <version>2.7.0</version>
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>2.7.0</version>
```

Comment:
I will emphasise that Spark can be **built** against different hadoop versions, but the release in maven central is a 2.2.0 build (latest). 
If Beam wants to support as many engines as possible (and we want it too), Beam needs to support Hadoop profiles to build against - i.e., the Spark runner will build against 2.2.0 by default.
This means that HadoopIO should probably support several hadoop version implementations.